distributed/comm/tests/test_ucx.py

```diff
@@ -170,7 +170,12 @@ async def test_ucx_deserialize():
         lambda cudf: cudf.DataFrame([1]).head(0),
         lambda cudf: cudf.DataFrame([1.0]).head(0),
         lambda cudf: cudf.DataFrame({"a": []}),
-        lambda cudf: cudf.DataFrame({"a": ["a"]}).head(0),
+        pytest.param(
+            lambda cudf: cudf.DataFrame({"a": ["a"]}).head(0),
+            marks=pytest.mark.skip(
+                reason="This test segfaults for some reason. So skip running it entirely."
+            ),
+        ),
```

Comment:
Not sure why this test now segfaults for me. Should note it does this reliably. However it wasn't that long ago when this test was skipped due to similar issues ( https://github.com/dask/distributed/pull/3667 ). So I guess it's not too surprising we are needing to skip it again. Still would be better if we got this to work instead.

Reply:
Looks like a zero sized slice of a StringColumn where it's possible that you're being given a non-nullptr with 0 size and that isn't handled properly?

Reply:
I'm not sure. When I debugged this with `pdb` locally, it appeared there was a non-trivial amount of data packed into a `DeviceBuffer`. However as soon as it got handed to UCX to send, it choked. Not entirely sure why.

Reply:
Ok have fixed this with the most recent commits.

Reply:
Added PR ( https://github.com/rapidsai/ucx-py/pull/521 ) to try and reproduce the error that was seen here.